### ISSUE TITLE
Cloudflare api token not working

### ISSUE THREAD

**Issue by kolaente** [containous/traefik#5592]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

I tried to use a scoped [cloudflare api token](https://api.cloudflare.com/#getting-started-endpoints) for the acme dns challenge as per the [documentation](https://go-acme.github.io/lego/dns/cloudflare/).

### What did you expect to see?

Challenge working.

### What did you see instead?

```
time="2019-10-07T19:23:17Z" level=error msg="Unable to obtain ACME certificate for domains \"vikunja.cloud,*.vikunja.cloud\" : cannot get ACME client cloudflare: some credentials information are missing: CLOUDFLARE_EMAIL,CLOUDFLARE_API_KEY"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.18
Codename:     maroilles
Go version:   go1.12.10
Built:        2019-09-26_01:56:30PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
debug = false
dashboard = true

logLevel = "ERROR"
defaultEntryPoints = ["https","http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  [entryPoints.https.tls]

  [entryPoints.traefik]
  address=":8080"
  [entryPoints.traefik.auth]
    [entryPoints.traefik.auth.basic]
      users = ["prom:PASSWORD"]

[metrics]
  [metrics.prometheus]
    entryPoint = "traefik"
    buckets = [0.1,0.3,1.2,5.0]

[retry]

[docker]
endpoint = "unix:///var/run/docker.sock"
watch = true
exposedByDefault = false

[acme]
email = "acme@example.com"
storage = "acme.json"
entryPoint = "https"
onHostRule = true
[acme.dnsChallenge]
  provider = "cloudflare"
  delayBeforeCheck = 0
[acme.httpChallenge]
entryPoint = "http"
[[acme.domains]]
  main = "example.com"
  sans = ["*.example.com"]
```

This is my docker-compose file with the cloudflare api token added:

```yaml
version: '2'

services:
  traefik:
    image: traefik:1.7
    restart: always
    ports:
      - 80:80
      - 443:443
    networks:
      - web
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./traefik.toml:/traefik.toml
      - ./acme.json:/acme.json
    environment:
      - CF_API_TOKEN=${CF_API_TOKEN}
    container_name: traefik

networks:
  web:
    external: true
```

**Comment by ldez** [comment]
Hello,

Currently Traefik doesn't support Cloudflare api token.

https://docs.traefik.io/https/acme/#providers

Also the Cloudflare api token support is not released in Lego, it will be released in v3.1.0

**Comment by kolaente** [comment]
Any eta when traefik will support it?

**Comment by bkeroackdsc** [comment]
It looks like this was released as part of 2.0.2

https://github.com/containous/traefik/pull/5607/files#diff-37aff102a57d3d7b797f152915a6dc16R42
